daemon/restart.go

```diff
@@ -39,8 +39,10 @@ func (daemon *Daemon) containerRestart(container *container.Container, seconds i
 		return err
 	}
 
-	if err := daemon.containerStart(container); err != nil {
-		return err
+	if !container.HostConfig.AutoRemove {
```

Comment:
I'm afraid setting restart policy to `always` won't work, if you `docker stop` a container with restart policy "always", it will stop and don't restart. This is strange, I would expect it will restart no matter how it is stopped.

I guess this is also limited by technique.